b12x/_lib/compiler.py

```
@@ -2430,112 +2797,300 @@ def _write_compile_manifest(
+        from cutlass.base_dsl.export.external_binary_module import (
+            ExternalBinaryModule,
+        )
+
+        # Load from the exact fd we wrote and verified — no name involved.
+        inode_path = f"/dev/fd/{stage_fd}"
+        module = ExternalBinaryModule(inode_path)
+        result = getattr(module, _cache_prefix(cache_key))
+        return result
+    except (RuntimeError, OSError, ValueError, json.JSONDecodeError):
+        return None
     except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)